### ISSUE TITLE
Bad data returned from control commands

### ISSUE THREAD

**Issue by leviathan747** [mdetools/mdetools18#2]
I'm consistently getting the return string `Rover,;` form my `Rover,getCompass()` and `Rover,GPS()` API calls. Is anyone else running into this?

**Comment by JuergenDingel** [comment]
Have you issued the 'ready' command before? Also, which OS are you using?

**Comment by leviathan747** [comment]
I am using macOS High Sierra.

Yes. Speaking of the 'ready' command, I am receiving an unexpected reply from this command with the string "Simulation Started". I am reading this reply from the socket and then discarding it...

Here is a relevant snippet from the part of my application that interfaces with the simulation:
```
 99     @Override              
100     public void pollRoverGPS( ComponentInstance_c senderReceiver ) {
101         String reply = rover.send( "Rover,GPS()" );
102         CorePlugin.out.println( reply );
103         float x = 0;       
104         float z = 0;
105         try {              
106             Matcher matcher = replyPattern.matcher( reply );
107             if ( matcher.matches() ) {          
108                 x = Float.parseFloat( matcher.group( 2 ) );
109                 z = Float.parseFloat( matcher.group( 4 ) );                                                                                                                          
110             }
111             else CorePlugin.err.println( "Match error in 'pollRoverGPS'" );                                                                                                          
112         }
113         catch ( NullPointerException e ) {}
114         catch ( NumberFormatException e ) {}
115         location_data_to_provider.roverGPS( null, x, z );
116     }
```

On line 101, `rover` is an instance of the `SocketCommunicator` class which I copied almost directly from the provided controller source with minor modification. On line 106, `replyPattern` is a regex `Pattern` object which is used to match the reply format specified in the instructions. This same pattern is used for all of the API calls which return values. The "Leader,GPS()" and "Leader,Distance()" invocations consistently return good data, however the "Rover,GPS()" and "Rover,getCompass()" invocations were causing an error to be printed (line 111). On further inspection, I saw that the string that was being returned was "Rover,;" with no values.

**Comment by iwaschosen** [comment]
A "Rover,;" message indicates a null return. I managed to identify an issue in the simulator which would cause a null return to be printed to the socket after receiving certain Power commands, which should not give any return message. The "Rover,;" message then waits in the socket until you read it. This would occur right after the next message which expects a return, such as getCompass or GPS. 

I updated the simulator to prevent these null messages from being printed to the socket, re-downloading it from the main page should fix the issue. Let me know if this works!
